blazar/plugins/oshosts/host_plugin.py

```diff
@@ -973,6 +973,46 @@ def poll_resource_failures(self):
                 recovered_hosts.extend([host for host in unreservable_hosts
                                         if host['id'] in active_hv_ids])
 
+            aggregates = self.nova.aggregates.list()
+            # create a map to get the current aggregate of a host in nova
+            host_to_agg_map = {host: agg for agg in aggregates for host in agg.hosts}
+            # get all physical hosts
+            all_hosts = db_api.host_list()
+            pool = nova.ReservationPool()
+            freepool = pool.get_aggregate_from_name_or_id(pool.freepool_name)
+            if not freepool:
+                raise ValueError("Aggregate list does not contain a freepool!")
+            for host in all_hosts:
+                # get the most recent reservation for the host_id and check if we need to move to freepool
+                reservation = db_utils.get_recent_non_pending_reservation_by_host_id(host['id'])
+                if reservation and reservation["reservation_status"] == status.reservation.ACTIVE:
+                    LOG.info(f"{host['hypervisor_hostname']} is in a active reservation"
+                             f"{reservation['reservation_id']}")
+                    continue
+                # host not in 'active' or 'pending' reservation should be in freepool
```

Comment:
this makes sense, but is a little bit confusing because we filter out "ACTIVE" reservations with the `continue`, and filter out `PENDING` reservations in the DB query. Would it be cleaner to filter out both active and pending reservations in the DB query?